File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;

int a[27], flag;
string s, ret;
char mid;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    cin >> s;
    for (char c: s) a[c-'A']++;

    for (int i=26; i>=0; i--) {
        if (a[i]) {
            if (a[i] & 1) {
                mid = char(i + 'A');
                a[i]--;
                flag++;
            }

            if (flag == 2) break;

            for (int j=0; j<a[i]; j+=2) {
                ret = char(i+'A') + ret + char(i+'A');
            }
        }
    }
    if (mid) ret.insert(ret.begin() + ret.size()/2, mid);

    if (flag == 2) cout << "I'm Sorry Hansoo";
    else cout << ret << '\n';

    return 0;
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
words = input()

alpha = [0] * 26
for w in words:
    alpha[ord(w) - ord('A')] += 1

flag = 0
ret = ''
mid = ''
for i in range(26):
    if alpha[i] % 2 == 1:
        mid = chr(i + ord('A'))
        flag += 1

    if flag > 1:
        print("I'm Sorry Hansoo")
        exit()

for i in range(26):
    ret += chr(i + ord('A')) * (alpha[i] // 2)

ret = ret + mid + ret[::-1]

print(ret)